main/lsp/requests/completion.cc

```diff
@@ -483,6 +484,38 @@ unique_ptr<CompletionItem> getCompletionItemForLocalName(const core::GlobalState
     return item;
 }
 
+vector<core::NameRef> fieldsForClass(LSPTypecheckerDelegate &typechecker, const core::ClassOrModuleRef klass,
+                                     const core::Loc queryLoc, ast::UnresolvedIdent::Kind kind) {
+    const auto &gs = typechecker.state();
+    auto files = vector<core::FileRef>{};
+    for (auto loc : klass.data(gs)->locs()) {
+        files.emplace_back(loc.file());
+    }
+    auto resolved = typechecker.getResolved(files);
+
+    // Instantiate fieldFinder outside loop so that result accumualates over every time we TreeMap::apply
+    FieldFinder fieldFinder(klass, queryLoc, kind);
```

Comment:
This doesn't look right. The reason why we have to use a tree walk for local variables is because there isn't a list of all local variables on the symbol table.

When we have an `enclosingMethod` (which we get from the `IdentResponse`), we can easily ask `enclosingMethod.data(ctx)->owner` for all of its fields. This would look much the same as it does today for `allSimilarConstantItems` (basically the same recursive algorithm) and would fix the problem that you're having that the current implementation doesn't handle instance variables declared in parent classes.

Reply:
Searching through the symbol table has the shortcoming that instance/class variables that are not typed don't get entered.  So, e.g. if you take this PR and run:

`./bazel-bin/main/sorbet -print symbol-table test/testdata/lsp/completion/instance_var.rb |grep field`

you get nothing.

I guess we could adopt a hybrid scheme, where files `typed: strict` and above get an implementation that searches through the symbol table (and is therefore exact) whereas files `typed: true` and below get purely syntactical completion results, which is not great, but is better than nothing?

Reply:
yep. i'm fine taking whatever approach to the variables that haven't been declared, but was just hoping that we could at least get all the ones we for sure knew about.

if we want i'm happy to think about approaches to get more undeclared instance variables into the symbol table.